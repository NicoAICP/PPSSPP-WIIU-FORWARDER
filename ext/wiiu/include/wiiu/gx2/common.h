#pragma once

#define GX2_SCAN_BUFFER_ALIGNMENT      0x1000
#define GX2_SHADER_ALIGNMENT           0x100
#define GX2_CONTEXT_STATE_ALIGNMENT    0x100
#define GX2_UNIFORM_BLOCK_ALIGNMENT    0x100
#define GX2_DISPLAY_LIST_ALIGNMENT     0x20
#define GX2_VERTEX_BUFFER_ALIGNMENT    0x40
#define GX2_INDEX_BUFFER_ALIGNMENT     0x20

#define GX2_ENABLE                     TRUE
#define GX2_DISABLE                    FALSE

#define GX2_TRUE                       TRUE
#define GX2_FALSE                      FALSE

#ifndef GX2_COMP_SEL
#define GX2_COMP_SEL(c0, c1, c2, c3) (((c0) << 24) | ((c1) << 16) | ((c2) << 8) | (c3))
#ifdef __cplusplus
#include <functional>
#endif
#include <stdio.h>

#define _x 0
#define _y 1
#define _z 2
#define _w 3
#define _r 0
#define _g 1
#define _b 2
#define _a 3
#define _0 4
#define _1 5
#endif


#define GX2_DISABLE_WRAPS
