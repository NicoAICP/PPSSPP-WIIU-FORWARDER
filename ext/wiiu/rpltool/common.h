#ifndef COMMON_H
#define COMMON_H

#include <ctype.h>
#include <stdint.h>
#include <stdio.h>

typedef int8_t s8;
typedef int16_t s16;
typedef int32_t s32;

typedef uint8_t u8;
typedef uint16_t u16;
typedef uint32_t u32;

#define GET_ENUM_NAME(prefix, name, val) \
   case prefix##name: \
      return #name;

#define EMIT_ENUM_TO_STR_FUNC(name, cases) \
   static inline const char *name##_to_str(name type) { \
      switch (type) { cases } \
      return "unknown"; \
   }

#define GET_FLAG_NAME(prefix, name, val) \
   case prefix##name: \
      pos += snprintf(buffer + pos, sizeof(buffer) - pos, "%s%s", #name, seperator); \
      flags &= ~(1u << i); \
      break;

#define EMIT_FLAGS_TO_STR_FUNC(name, cases) \
   static inline const char *name##_to_str(int flags, const char *seperator) { \
      int i; \
      static char buffer[32 * 32]; \
      int pos = 0; \
      for (i = 0; i < 32; i++) \
         if ((1u << i) & flags) \
            switch (1u << i) { cases } \
      if (flags) \
         snprintf(buffer + pos, sizeof(buffer) - pos, "0x%08X", flags); \
      else if (pos) \
         buffer[pos - strlen(seperator)] = '\0'; \
      else \
         return "NULL"; \
      return buffer; \
   }

#define MAKE_MAGIC(a, b, c, d) (((a) << 24) | ((b) << 16) | ((c) << 8) | ((d) << 0))

#define log_var(X) printf("%-35s: 0x%0*X (%u)\n", #X, sizeof(X) * 2, (u32)(X), (u32)(X))
#define log_svar(X) printf("%-35s: 0x%0*X (%i)\n", #X, sizeof(X) * 2, (u32)(X), (s32)(X))
#define log_array(X) debug_array(X, sizeof(X), #X)
#define log_enum(type, X) printf("%-35s: 0x%X (%s)\n", #X, (X), type##_to_str(X))
#define log_flags(type, X) printf("%-35s: 0x%08X (%s)\n", #X, (X), type##_to_str(X, " | "))
#define log_magic(X) \
   printf("%-35s: '%c''%c''%c''%c' (0x%08X)\n", #X, (u32)(X) >> 24, (u32)(X) >> 16, (u32)(X) >> 8, (u32)(X), (u32)(X))
#define log_carray(X) debug_char_array(X, sizeof(X), #X)
#define log_str(X) printf("%-35s: \"%s\"\n", #X, (char *)(X))
#define log_off_str(X, data) printf("%-35s: 0x%08X \"%s\"\n", #X, (X), (char *)data + (X))
#define log_off_strs(X, data) \
   do { \
      printf("%-35s: 0x%08X ", #X, (X)); \
      const char *ptr = (char *)data + (X); \
      while (*ptr) { \
         ptr += printf("\"%s\" ", ptr) - 2; \
      } \
      printf("\n"); \
   } while (0)

static inline void dump_array(uint32_t *src, int size);
static inline void debug_array(uint32_t *src, int size, const char *name) {
   printf("%-35s:", name);
   dump_array(src, size);
}
static inline void debug_char_array(uint8_t *src, int size, const char *name) {
   int i;
   printf("%-35s: ", name);
   //   for (i = 0; i < size; i++)
   //      printf("%-2c ", isalnum(src[i])?src[i]: '?');
   //   printf("\n%-35s: ", "");
   for (i = 0; i < size; i++)
      printf("%02X ", src[i]);
   printf("\n");
}

static inline void dump_array(uint32_t *src, int size) {
   int i;
   size = (size / 4) - 1;
   printf("\n{");
   for (i = 0; i < size; i++) {
      if (!(i & 3))
         printf("\n   ");

      printf("0x%08X, ", src[i]);
   }

   if (!(i & 3))
      printf("\n   ");

   printf("0x%08X\n}\n", src[i]);
}

static inline u32 align_up(u32 x, u32 align) { return (x + (align - 1)) & ~(align - 1); }
static inline u32 align_down(u32 x, u32 align) { return x & ~(align - 1); }

#endif // COMMON_H
