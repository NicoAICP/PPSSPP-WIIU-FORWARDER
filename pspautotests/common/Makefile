TARGET_LIB = libcommon.a
EXTRA_TARGETS = libcommon_kernel.a
OBJS = common.o vram.o sysmem-imports.o
KERNEL_OBJS = common_kernel.o vram.o sysmem-imports.o

INCDIR = 
CFLAGS = -O0 -G 0 -g -Wall
CXXFLAGS = $(CFLAGS) -fno-exceptions -fno-rtti
ASFLAGS = $(CFLAGS)

LIBDIR =
LDFLAGS = -G0

PSPSDK=$(shell psp-config --pspsdk-path)
include $(PSPSDK)/lib/build.mak

%.o: %.S
	$(AS) $(ASFLAGS) -c -o $@ $<

common_kernel.o: common.c
	$(CC) $(CFLAGS) -DCOMMON_KERNEL=1 -c -o $@ $<

libcommon_kernel.a: $(KERNEL_OBJS)
	$(AR) cru $@ $(KERNEL_OBJS)
	$(RANLIB) $@
