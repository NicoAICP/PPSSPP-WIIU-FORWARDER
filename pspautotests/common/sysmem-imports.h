extern int sceKernelSetCompiledSdkVersion(int ver);
extern int sceKernelSetCompiledSdkVersion370(int ver);
extern int sceKernelSetCompiledSdkVersion380_390(int ver);
extern int sceKernelSetCompiledSdkVersion395(int ver);
extern int sceKernelSetCompiledSdkVersion401_402(int ver);
extern int sceKernelSetCompiledSdkVersion500_505(int ver);
extern int sceKernelSetCompiledSdkVersion507(int ver);
extern int sceKernelSetCompiledSdkVersion600_602(int ver);
extern int sceKernelSetCompiledSdkVersion603_605(int ver);
extern int sceKernelSetCompiledSdkVersion606(int ver);
extern int sceKernelGetCompiledSdkVersion();