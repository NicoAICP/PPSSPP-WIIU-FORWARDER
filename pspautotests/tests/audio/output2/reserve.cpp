#include <common.h>
#include <pspaudio.h>
#include <pspthreadman.h>

static void test2Reserve(const char *title, int samples) {
	int result = sceAudioOutput2Reserve(samples);
	if (result >= 0) {
		sceAudioOutput2Release();
	}
	checkpoint("%s: %08x", title, result);
}

static void testSRCReserve(const char *title, int samples, int freq, int fmt) {
	int result = sceAudioSRCChReserve(samples, freq, fmt);
	if (result >= 0) {
		sceAudioSRCChRelease();
	}
	checkpoint("%s: %08x", title, result);
}

extern "C" int main(int argc, char *argv[]) {
	checkpointNext("Output2");
	sceAudioOutput2Reserve(17);
	checkpoint("  Twice: %08x", sceAudioOutput2Reserve(17));
	sceAudioOutput2Release();

	test2Reserve("  Zero", 0);
	test2Reserve("  Negative", -1);
	static const int sampleCounts[] = { 16, 17, 18, 2048, 2049, 4110, 4111, 4112, 8222, 0x80000000 | 4111, 0x40000000 | 4111, 0x80000000 | 16, 0x80000000 | 17 };
	for (size_t i = 0; i < ARRAY_SIZE(sampleCounts); ++i) {
		char temp[256];
		sprintf(temp, "  %d samples", sampleCounts[i]);
		test2Reserve(temp, sampleCounts[i]);
	}

	checkpointNext("SRC");
	sceAudioSRCChReserve(17, 44100, 2);
	checkpoint("  SRC Twice: %08x", sceAudioSRCChReserve(17, 44100, 2));
	sceAudioSRCChRelease();

	checkpointNext("SRC samples");
	testSRCReserve("  Zero SRC samples", 0, 44100, 2);
	testSRCReserve("  Negative SRC samples", -1, 44100, 2);
	for (size_t i = 0; i < ARRAY_SIZE(sampleCounts); ++i) {
		char temp[256];
		sprintf(temp, "  %d SRC samples", sampleCounts[i]);
		testSRCReserve(temp, sampleCounts[i], 44100, 2);
	}

	checkpointNext("SRC frequencies");
	testSRCReserve("  Zero frequency", 64, 0, 2);
	testSRCReserve("  Negative frequency", 64, -1, 2);
	static const int freqs[] = { 1, 48000, 44100, 32000, 24000, 22050, 16000, 12000, 11025, 8000, 4000, 40000, 88200, 64000, 360000, 0x80000000 | 44100 };
	for (size_t i = 0; i < ARRAY_SIZE(freqs); ++i) {
		char temp[256];
		sprintf(temp, "  %d frequency", freqs[i]);
		testSRCReserve(temp, 64, freqs[i], 2);
	}

	checkpointNext("SRC formats");
	testSRCReserve("  Zero format", 64, 0, 0);
	testSRCReserve("  Negative format", 64, 0, -2);
	static const int formats[] = { 1, 2, 3, 4, 5, 8, 0x80000002 };
	for (size_t i = 0; i < ARRAY_SIZE(formats); ++i) {
		char temp[256];
		sprintf(temp, "  %d format", formats[i]);
		testSRCReserve(temp, 64, 0, formats[i]);
	}

	return 0;
}