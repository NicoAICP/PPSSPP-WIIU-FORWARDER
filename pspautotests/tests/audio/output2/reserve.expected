[x] Output2
[x]   Twice: 80268002
[x]   Zero: 80000104
[x]   Negative: 80000104
[x]   16 samples: 80000104
[x]   17 samples: 00000000
[x]   18 samples: 00000000
[x]   2048 samples: 00000000
[x]   2049 samples: 00000000
[x]   4110 samples: 00000000
[x]   4111 samples: 00000000
[x]   4112 samples: 80000104
[x]   8222 samples: 80000104
[x]   -2147479537 samples: 00000000
[x]   1073745935 samples: 80000104
[x]   -2147483632 samples: 80000104
[x]   -2147483631 samples: 00000000

[x] SRC
[x]   SRC Twice: 80268002

[x] SRC samples
[x]   Zero SRC samples: 80000104
[x]   Negative SRC samples: 80000104
[x]   16 SRC samples: 80000104
[x]   17 SRC samples: 00000000
[x]   18 SRC samples: 00000000
[x]   2048 SRC samples: 00000000
[x]   2049 SRC samples: 00000000
[x]   4110 SRC samples: 00000000
[x]   4111 SRC samples: 00000000
[x]   4112 SRC samples: 80000104
[x]   8222 SRC samples: 80000104
[x]   -2147479537 SRC samples: 00000000
[x]   1073745935 SRC samples: 80000104
[x]   -2147483632 SRC samples: 80000104
[x]   -2147483631 SRC samples: 00000000

[x] SRC frequencies
[x]   Zero frequency: 00000000
[x]   Negative frequency: 8026000a
[x]   1 frequency: 8026000a
[x]   48000 frequency: 00000000
[x]   44100 frequency: 00000000
[x]   32000 frequency: 00000000
[x]   24000 frequency: 00000000
[x]   22050 frequency: 00000000
[x]   16000 frequency: 00000000
[x]   12000 frequency: 00000000
[x]   11025 frequency: 00000000
[x]   8000 frequency: 00000000
[x]   4000 frequency: 8026000a
[x]   40000 frequency: 8026000a
[x]   88200 frequency: 8026000a
[x]   64000 frequency: 8026000a
[x]   360000 frequency: 8026000a
[x]   -2147439548 frequency: 8026000a

[x] SRC formats
[x]   Zero format: 80000104
[x]   Negative format: 80000104
[x]   1 format: 80000104
[x]   2 format: 00000000
[x]   3 format: 80000104
[x]   4 format: 80000003
[x]   5 format: 80000104
[x]   8 format: 80000104
[x]   -2147483646 format: 80000104
